### PR TITLE
Repository Overhaul

### PR DESCRIPTION
**Pull request by ElectronicsArchiver**
[Badge Suggestions]: https://img.shields.io/badge/Ｒｅｐｏｓｉｔｏｒｙ_Ｓｕｇｇｅｓｔｉｏｎｓ-609926?style=for-the-badge&logoColor=white&logo=GitHub
[Badge Changes]: https://img.shields.io/badge/Ｃｈａｎｇｅｓ-D81939?style=for-the-badge&logoColor=white&logo=GitLFS
[Badge Ask]: https://img.shields.io/badge/Ｐｌｅａｓｅ_Ａｓｋ_Ｍｅ-73398D?style=for-the-badge&logoColor=white&logo=AskUbuntu

[Widgets]: https://github.com/MarkedDown/Disable/blob/main/Features/Widgets.md
[Tabs]: https://github.com/MarkedDown/Disable/blob/main/Features/Tabs.md

<br>
<br>

<div align = center>

[<kbd> <br> Please Check The **Preview** <br> </kbd>][Preview]

</div>
  
<br>
<br>

## ![Badge Suggestions]

  <br>

- ***[Disable Unused Tabs][Tabs]***:

  - `Projects`

  - `Wiki`

<br>
<br>

## ![Badge Changes]

- Sorted files

- Fixed paths / workflows

  *Please check*

- Formatted / reworked README + documentation

<br>
<br>

## ![Badge Ask]

- ***If you have any questions.***

- ***In case you do not agree with some of the changes made in this PR***, <br>
  *please first let me know about them, before disregarding it entirely so*<br>
  *we can possibly find a middle ground and have it not go to waste.*

<br>
<br>

[Preview]: https://github.com/ElectronicsArchiver/hotdog

### CODE REVIEW

File: .github/workflows/release.yml
Comment (Sauceke) [inline .github/workflows/release.yml]: This would also affect the release asset URL and break links to it. Please keep the old name.
```suggestion
        run: cp Source/Firmware/hotdog-fw.ino assets
```

File: Documentation/Wired.md
Comment (Sauceke) [inline Documentation/Wired.md]: I don't see the benefit of these line breaks, md pages in github are already responsive by design.

File: Documentation/Wired.md
Comment (Sauceke) [inline Documentation/Wired.md:3]: Unnecessary blank line here and throughout the PR.

File: Documentation/Wired.md
Comment (Sauceke) [inline Documentation/Wired.md:6]: I'd prefer to let the engine decide line spacing; using `<br>` for spacing is an anti-pattern anyway. Similar comment on the other md files.

File: Documentation/Wireless.md
Comment (Sauceke) [inline Documentation/Wireless.md]: I think setting all images to the same height makes it look more organized. Why did you decide to change the layout?

File: Documentation/Wired.md
Comment (Sauceke) [inline Documentation/Wired.md:19]: I prefer no blank lines between list items either (and keeping list items to a single line when possible), mostly for readability when viewed as plain text.

File: Documentation/README.md
Comment (Sauceke) [inline Documentation/README.md]: These links are pretty important, I'd move them back to the top-level readme page.

File: Documentation/Wireless.md
Comment (Sauceke) [inline Documentation/Wireless.md:97]: Uh
Why COM0 specifically?

**Review by Sauceke — CHANGES_REQUESTED** [review]

File: Documentation/README.md
Comment (ElectronicsArchiver) [reply Documentation/README.md]: Sure

File: Documentation/Wireless.md
Comment (ElectronicsArchiver) [reply Documentation/Wireless.md:97]: `COM0` is the default USB port on Windows systems.

File: .github/workflows/release.yml
Comment (ElectronicsArchiver) [reply .github/workflows/release.yml]: Since you use the Arduino IDE and not something like PlatformIO where 
changing a folders name is no issue, I adjusted the firmwares filename.

Couldn't I just update the link with your next release?

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md]: The benefit is their functionality, additional spacing.

GitHub is 'responsive' in that it does some responsive things,
not that it does everything / does everything well.

Vertical separation is quite impactful.

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md:3]: They aren't unnecessary, they make code - in this case markdown -
readable and not a cluttered mess of shoulder on shoulder sections.

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md:6]: As markdown is an extension of html and not 
a subset, not utilizing it to improve redability 
seems quite counterintuitive to me.

File: Documentation/Wireless.md
Comment (ElectronicsArchiver) [reply Documentation/Wireless.md]: Because it looked less organized without it.

The images aren't all of the same size, as such image pairs
that use the same width instead of the same height can look
rather randomly placed.

If you mean that singular images should be as 
tall as the image pairs then I'd agree with you.

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md:19]: How is one stuffed line more readable
then an aligned paragraph though?

File: .github/workflows/release.yml
Comment (Sauceke) [inline .github/workflows/release.yml]:
```suggestion
        run: cp Source/Firmware/Firmware.ino assets/hotdog-fw.ino
```

Comment (Sauceke) [reply .github/workflows/release.yml]: Ah, true
Then could you please do [this](https://github.com/Sauceke/hotdog/pull/2#discussion_r912049697) instead

Comment (ElectronicsArchiver) [reply .github/workflows/release.yml]: Not sure why, but sure?

Comment (ElectronicsArchiver) [reply .github/workflows/release.yml]: Ah, ok, you comment in the other one.

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md]: It is conventional for text documents to use the available horizontal space (see any news site, blog, Wikipedia etc.), and that is what most people are used to.
If you don't like long lines, then please at least add the line breaks along logical boundaries (e. g. between sentences).

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md:3]: Okay, accepted.

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md]: > see any news site, blog

Those are examples that usually don't do this as they 
understand, as newspapers / books / papers /.. do, 
that the human perception limit is about 60 characters.

I usually go with about 60 - 80 as a compromise.

> If you don't like long lines, then please at least add the line breaks along logical boundaries (e. g. between sentences).

Well since GitHub isn't really responsive, it doesn't have 
the required functionality of the `justify` alignment type 
which is required, if you want to cut sentences at length 
wise random positions.

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md:6]: We don't use br for spacing in HTML either. We use margins, and GH adds those for you automatically. Besides, I honestly think this is too much spacing.

Comment (ElectronicsArchiver) [reply Documentation/Wired.md:6]: > We don't use br for spacing in HTML either

Who is we?

Either way, that doesn't change the fact that 
it is default functionality of HTML, nor that it 
is automatically added to **every** textfield that 
is modified with javascript and set to text with 
multiple parts.

File: Documentation/Wireless.md
Comment (Sauceke) [reply Documentation/Wireless.md]: Yes, that's what I mean, but that's not what I'm seeing. E.g. R1-R5_mount.jpeg is much smaller (both in width and in height) than the images before and after it. Was that intentional on your part or just a typo?

Comment (ElectronicsArchiver) [reply Documentation/Wireless.md]: > Was that intentional on your part or just a typo?

In that case no, I just didn't adjust 
the singular images, I left them with 
the size you had set them to.

I'll adjust them then~

Comment (ElectronicsArchiver) [reply Documentation/Wireless.md]: I adjusted the images to be relatively 
sized to their neighbours.

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md:6]: > Who is we?

[Just about everyone](https://www.google.com/search?q="using+br+for+spacing"), it seems.

Look, I'm not going to force you to do things that you ideologically disagree with, but if I find any of these tags after merging, I'll just remove them myself :)

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md:19]: I wouldn't call a line of just six words "stuffed". Some haikus have longer lines than that.

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md]: 80 sounds good to me, but please break along full stops whenever possible. These pages aren't rendered in a monospaced font anyway, so breaking along a given character limit isn't that meaningful IMHO.

File: Documentation/Wireless.md
Comment (Sauceke) [reply Documentation/Wireless.md:97]: Weird, it doesn't show up for me no matter which port I try. Well, whatever.

File: Documentation/Wireless.md
Comment (Sauceke) [reply Documentation/Wireless.md]: Thank you, looks much better.

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md:6]: > [Just about everyone](https://www.google.com/search?q=%22using+br+for+spacing%22), it seems.

Not sure what results you are talking about..

They are personalized to begin with, I 
might see something rather different idk?

> We use margins

Well you don't have those on GitHub though.

- *As a user obs*
- *Except for GitHub pages*

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md:19]: It's not about the word count, but the fact 
that it's two separate things in one line.

### Guides

To begin with, I would generally reorder
the assembly guides into multiple parts
and not one long one.

Including creating a dedicated 
section for 'things I used'.

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md:6]: > Well you don't have those on GitHub though.

What I meant to say, is that github automatically sets the margins for h1, h2 etc. and that I'm perfectly happy with those defaults. They don't feel cluttered to me.

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md]: > so breaking along a given character limit isn't that meaningful IMHO.

I know, I try to make lines visually have the same 
length, as this also helps readability, however this 
usually requires me to reword sentences multiple 
times until I find the 'perfect' wording, which I didn't 
do much in this PR.

In intial PRs of bigger repos I usually first 
go for basic groundwork ( which is still a lot 
of work ) as they might very well get denied 
outright, sometimes without even a reason.

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md:6]: Right.

> but if I find any of these tags after merging, I'll just remove them myself :)

Well if you will still merge it, I'm fine with it.

That way you can check for yourself
how it looks while you remove them.

If you use Atom, the live markdown 
preview package works quite nicely.

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md]: I appreciate the effort, but please keep in mind, that now I'll have to reindent the lines each time I edit these documents, which is not necessarily something I'll enjoy doing. Which is why I have a strong preference to keep the default indentation, or at least to have it split into manageable chunks (such as sentences), and not something that will cascade through 10 lines.

File: Documentation/Wired.md
Comment (Sauceke) [reply Documentation/Wired.md:19]: Not important enough to further argue about. Accepted.

File: Documentation/Wired.md
Comment (ElectronicsArchiver) [reply Documentation/Wired.md]: > I appreciate the effort, but please keep in mind, that now I'll have to reindent the lines each time I edit these documents, which is not necessarily something I'll enjoy doing. 

Huh, any modern editor should follow the indentation 
level of the previous line / line with content though..

I'm a bit confused how you came to indentation level
as this `request changes` conversation was about line
length..

In case you meant line length and are concerned about
having to keep the visual length equal from section to
section then just don't.

Having many small blocks with slightly different block
widths looks fine as long as the block / sentence is 
broken normally.

Not that you have to do it properly yourself anyways,
you could just half hazardly break them wherever and
have them be cleaned up by me in some future PR.

Comment (Sauceke) [reply Documentation/Wired.md]: I mean line width - sorry for the confusion.
Fine. Keep the line breaks, but please use the proper Markdown syntax, which is two spaces. Not having any of this "haha readme go `<br>`" menace.
As an aside, most browsers (Chrome, Firefox, Edge) come with a reader mode that does pretty much the same thing, so I still find line breaking by hand redundant.

Comment (ElectronicsArchiver) [reply Documentation/Wired.md]: > I mean line width - sorry for the confusion.

No problem~

> but please use the proper Markdown syntax, which is two spaces

Are you referring to vertical spacing,
maybe I'm just stupid but I'm not sure
what you mean.

> As an aside, most browsers (Chrome, Firefox, Edge) come with a reader mode that does pretty much the same thing, so I still find line breaking by hand redundant.

Well most people don't use it.

If it's not automatically used
not many people will use it to
begin with..

Btw, thankfully Edge is dead~

Comment (Sauceke) [reply Documentation/Wired.md]: https://www.markdownguide.org/basic-syntax#line-breaks

Comment (ElectronicsArchiver) [reply Documentation/Wired.md]: Mhmhmh, very interesting indeed.

Tried it out, very cool~

<br>

Though since markdown doesn't highlight trailing spaces 
and many linting tools automatically remove them I'm not 
sure if I'll prefer them in the future.

<br>

***I'll adjust the PR then~***

File: Documentation/Wireless.md
Comment (Sauceke) [inline Documentation/Wireless.md]:
```suggestion
    I prefer to also bend the wires in before clipping  
```

**Review by Sauceke — CHANGES_REQUESTED** [review]

**Review by Sauceke — APPROVED** [review]
Great quality work. Thank you.
Some recommendations to get your future contributions accepted more easily:
- Keep PRs atomic. A good pull request addresses exactly one issue. Large PRs are hard to review, and stricter maintainers will reject them outright. I would recommend one PR for refactoring, one for changing the content, and one for reformatting.
- Your PR description feels generated, and it wasn't clear to me that you're not a bot until further inspection. Using sentences instead of bullet points would be better IMO.

Keep up the good work.

**Comment by ElectronicsArchiver** [comment]
> Great quality work. Thank you.

:+1: 

> Keep PRs atomic. A good pull request addresses exactly one issue. Large PRs are hard to review, and stricter maintainers will reject them outright. I would recommend one PR for refactoring, one for changing the content, and one for reformatting.

I know. It depends, but yes, doing it properly would be making multiple PRs~

> Your PR description feels generated, and it wasn't clear to me that you're not a bot until further inspection. Using sentences instead of bullet points would be better IMO.

Doesn't help that I have a gundam related avatar x3
Many people probably think that I'm some sort of `dependabot` like tool.

Actually at least 5 peoples first response to my PRs was
`Not sure if you are an automated ...`

**Pull request merged**
